rasp/tor_engine.py

```diff
@@ -44,29 +45,13 @@ def send_signal(self):
 
     def get_page_source(self, url):
         try:
-            if url:
-                self.send_signal()
-                req = urllib.request.Request(url, self.data, self.headers)
-                res = urllib.request.urlopen(req)
-                if res:
-                    try:
-                        return Webpage(url, str(res.read()))
-                    except http.client.IncompleteRead:
-                        return None
-                else:
-                    return None
-            else:
-                return None
+            self.send_signal()
+            req = urllib.request.Request(url, self.data, self.headers)
+            res = urllib.request.urlopen(req)
```

Comment:
We should add the response code into the Webpage object and return that, with status code 500, and no source.